sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -218,15 +234,38 @@ function appendQueryParams(url: string, queryParams: Map<string, string | string
   // QUIRK: parsedUrl.searchParams will have their name/value pairs decoded, which
   // can change their meaning to the server, such as in the case of a SAS signature.
   // To avoid accidentally un-encoding a query param, we parse the key/values ourselves
-  const existingParams = simpleParseQueryParams(parsedUrl.search);
-  const combinedParams = new Map<string, string | string[]>(existingParams);
+  const combinedParams = simpleParseQueryParams(parsedUrl.search);
 
   for (const [name, value] of queryParams) {
     const existingValue = combinedParams.get(name);
     if (Array.isArray(existingValue)) {
-      existingValue.push(...value);
+      if (Array.isArray(value)) {
+        if (existingValue.toString() !== value.toString()) {
```

Comment:
I don't think this is a good way to do deep equality checking of arrays, especially since some types get coerced into being the empty string, e.g.

![image](https://user-images.githubusercontent.com/639216/122135348-2d8f9c00-cdf5-11eb-8240-eac890beb884.png)

If we want to de-dupe values, we should use a Set instead of an array to store the values.

Reply:
Modified the logic for comaparison.